frontend/src/components/StatisticsPage/Charts.jsx

```diff
@@ -1,27 +1,39 @@
 import React from "react";
-import { Line } from "react-chartjs-2";
+import { Line, Chart } from "react-chartjs-2";
 import { connect } from "react-redux";
 import { Header } from "semantic-ui-react";
 
-const chartColors = [
-  "red",
-  "orange",
-  "olive",
-  "green",
-  "teal",
-  "blue",
-  "violet",
-  "purple",
-  "pink",
-  "brown",
-  "grey",
-  "black",
-  "yellow"
-];
-
-const intToColor = i => {
-  return chartColors[i % chartColors.length];
-};
+let clicks = 0;
+let timeout;
+const delay = 200;
+
+let defaultOnClick = Chart.defaults.global.legend.onClick;
+
+const getRandomIntInclusive = (min, max) => {
+  min = Math.ceil(min);
+  max = Math.floor(max);
+  return Math.floor(Math.random() * (max - min + 1)) + min;
+}
+
+const shuffleArray = (array) => {
+  for(let i=0; i < array.length - 1; i++) {
+    const j = getRandomIntInclusive(0, array.length - 1);
+    [array[i], array[j]] = [array[j], array[i]];
+  }
+}
+
+const generateRandomColors = (n) => {
```

Comment:
Maybe extract this function to some util/helper file in case someone would use it elsewhere